### DISCUSSION TITLE
Error when using model fine-tuned with DocLayNet

### DISCUSSION THREAD

**Discussion by yflam1** [deepdoctection/deepdoctection#165] [Q&A] [answered]
Thank you for your quick response on the issue of CUDA OOM last time. The fine-tuning is done and the result looks promising. However, there is another issue now. I copied the fine-tuned model (`model_final.pth`) to `~/.cache/deepdoctection/weights/layout`, and registered the model:

```python
dd.ModelCatalog.register("layout/model_final.pth", dd.ModelProfile(
    name="layout/model_final.pth",
    description="layout_detection/doclaynet",
    config="dd/d2/layout/CASCADE_RCNN_R_50_FPN_GN.yaml",
    tp_model=False,
    size = [],
    categories={
        "1": dd.LayoutType.text,
        "2": dd.LayoutType.title,
        "3": dd.LayoutType.list,
        "4": dd.LayoutType.table,
        "5": dd.LayoutType.figure,
    },
    model_wrapper="D2FrcnnDetector",
))
```

Then, I changed my `ImageLayoutService` to:

```python
layout_config_path = dd.ModelCatalog.get_full_path_configs("dd/d2/layout/CASCADE_RCNN_R_50_FPN_GN.yaml")
layout_weights_path = dd.ModelCatalog.get_full_path_weights("layout/model_final.pth")
categories_layout = dd.ModelCatalog.get_profile("layout/model_final.pth").categories
d_layout = dd.D2FrcnnDetector(layout_config_path, layout_weights_path, categories_layout)
layout = dd.ImageLayoutService(d_layout)
pipe_component_list.append(layout)
```

Unfortunately, I get the following error:

```bash
Traceback (most recent call last):
  ...
  File ".../lib/python3.9/site-packages/deepdoctection/dataflow/common.py", line 109, in __iter__
    for dp in self.df:
  File ".../lib/python3.9/site-packages/deepdoctection/dataflow/common.py", line 109, in __iter__
    for dp in self.df:
  File ".../lib/python3.9/site-packages/deepdoctection/dataflow/common.py", line 109, in __iter__
    for dp in self.df:
  [Previous line repeated 5 more times]
  File ".../lib/python3.9/site-packages/deepdoctection/dataflow/common.py", line 110, in __iter__
    ret = self.func(copy(dp))  # shallow copy the list
  File ".../lib/python3.9/site-packages/deepdoctection/pipe/base.py", line 93, in pass_datapoint
    self.serve(dp)
  File ".../lib/python3.9/site-packages/deepdoctection/pipe/cell.py", line 185, in serve
    raise ValueError("sub_image_ann.image is None, but must be an image")
ValueError: sub_image_ann.image is None, but must be an image
```

When I switched back to the original `layout/d2_model_0829999_layout_inf_only.pt`, everything works. What could be the reasons?

**Comment by JaMe76** [comment, marked as answer]
Detected objects are internally saved as `ImageAnnotation` instance `image_annotation` with bounding boxes, category names etc. 

This error occurs when you have an instance `image_annotation` (i assume a table) that you crop and you want then to perform some detection operations at the cropped sub image. This is solved by creating an `Image` instance given by annotation bounding boxes and storing the instance at `image_annotation.image`.

Creating an `Image` instances from `ImageAnnotation` can be achieved setting `to_image=True` in `ImageLayoutService`. If you not only want to generate an `Image` instance but also want to generate a np array of the sub image you can set `crop_image=True`.

'''
layout = ImageLayoutService(d_layout, to_image=True, crop_image=True)
'''

Maybe the reason it worked with other layout mode is that the new model now detects a table?

**Reply by yflam1** [reply]
Ah. Thanks!
